WooCommerce/Classes/ViewRelated/ListSelector/ListSelectorViewController.swift

```diff
@@ -56,29 +60,37 @@ UIViewController, UITableViewDataSource, UITableViewDelegate where DataSource.Mo
     }
 
     override func viewWillDisappear(_ animated: Bool) {
-        onDismiss(dataSource.selected)
+        if isMovingFromParent {
+            onDismiss(command.selected)
+        }
         super.viewWillDisappear(animated)
     }
 
+    func reloadData() {
+        tableView.reloadData()
+
+        title = command.navigationBarTitle
```

Comment:
>  the navigation bar title is no longer constant

When does the title change? Is it on every tap? Or on data load only?

Reply:
the navigation bar title could change when the number of active filters changes, so on data load only